src/components/calcite-panel/calcite-panel.tsx

```diff
@@ -0,0 +1,202 @@
+import { Component, Element, Event, EventEmitter, Host, Prop, Watch, h } from "@stencil/core";
+import { CSS, TEXT } from "./resources";
+import { getElementDir } from "../utils/dom";
+import classnames from "classnames";
+import { CSS_UTILITY } from "../utils/resources";
+import { VNode } from "@stencil/core/dist/declarations";
+import { CalciteTheme } from "../interfaces";
+import CalciteIcon from "../utils/CalciteIcon";
+import { x16 } from "@esri/calcite-ui-icons";
+
+const SLOTS = {
+  headerContent: "header-content",
+  headerLeadingContent: "header-leading-content",
+  headerTrailingContent: "header-trailing-content",
+  footer: "footer"
+};
+
+/**
+ * @slot header-content - A slot for adding content in the center of the header.
+ * @slot header-leading-content - A slot for adding a `calcite-action` on the leading side of the header.
+ * @slot header-trailing-content - A slot for adding a `calcite-action` on the trailing side of the header.
+ * @slot footer - A slot for adding `calcite-actions` to the footer.
+ */
+@Component({
+  tag: "calcite-panel",
+  styleUrl: "calcite-panel.scss",
+  shadow: true
+})
+export class CalcitePanel {
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  /**
+   * Hides the panel.
+   */
+  @Prop({ mutable: true, reflect: true }) dismissed = false;
```

Comment:
This blog has as good comparison between the internal state vs mutable prop.

https://auth0.com/blog/creating-web-components-with-stencil/#Managing-Mutable-Properties-on-Stencil

The internal state way of doing it just requires more code.

What should we follow for our best practices @jcfranco ?

Reply:
Nice find! 

Honestly, I wouldn't want to enforce one way or another. There are benefits of separating state and props, but it *depends on the component*. If it can be done simply and cleanly with prop metadata why not?